Client/mods/deathmatch/logic/rpc/CElementRPCs.cpp

```diff
@@ -426,38 +426,73 @@ void CElementRPCs::SetElementHealth(CClientEntity* pSource, NetBitStreamInterfac
 void CElementRPCs::SetElementModel(CClientEntity* pSource, NetBitStreamInterface& bitStream)
 {
     unsigned short usModel;
-    if (bitStream.Read(usModel))
+
+    if (!bitStream.Read(usModel))
+        return;
+    
+    switch (pSource->GetType())
     {
-        switch (pSource->GetType())
+        case CCLIENTPED:
+        case CCLIENTPLAYER:
         {
-            case CCLIENTPED:
-            case CCLIENTPLAYER:
-            {
-                CClientPed* pPed = static_cast<CClientPed*>(pSource);
-                pPed->SetModel(usModel);
-                break;
-            }
+            CClientPed* pPed = static_cast<CClientPed*>(pSource);
+            const unsigned short usCurrentModel = pPed->GetModel();
 
-            case CCLIENTVEHICLE:
+            if (usCurrentModel != usModel)
             {
-                uchar ucVariant = 255, ucVariant2 = 255;
-                if (bitStream.GetNumberOfUnreadBits() >= sizeof(ucVariant) + sizeof(ucVariant2))
+                if (pPed->SetModel(usModel))
                 {
-                    bitStream.Read(ucVariant);
-                    bitStream.Read(ucVariant2);
+                    CLuaArguments Arguments;

```

Comment:
Here you trigger the event only if `SetModel` returns `true` for peds but you don't do that check for other types.
Is that intended ?

Reply:
Only `CClientPed::SetModel` returns a boolean.
https://github.com/multitheftauto/mtasa-blue/blob/14a5c3755b14512819ecd4bdb89f3161d79316ca/Client/mods/deathmatch/logic/CClientPed.h#L209
https://github.com/multitheftauto/mtasa-blue/blob/14a5c3755b14512819ecd4bdb89f3161d79316ca/Client/mods/deathmatch/logic/CClientVehicle.h#L209
https://github.com/multitheftauto/mtasa-blue/blob/14a5c3755b14512819ecd4bdb89f3161d79316ca/Client/mods/deathmatch/logic/CClientObject.h#L70
https://github.com/multitheftauto/mtasa-blue/blob/14a5c3755b14512819ecd4bdb89f3161d79316ca/Client/mods/deathmatch/logic/CClientProjectile.h#L85

Reply:
Fair enough.